### ISSUE TITLE
Enable Autocorrect Option in Xcode Build Tool Plugin

### ISSUE THREAD

**Issue by StefaniOSApps** [realm/SwiftLint#5378]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I am currently utilizing SwiftLint as a Build Tool Plugin in Xcode. While the linting functionality works as expected, I noticed that the autocorrect feature is not available in this integration mode. This seems to limit the usability of SwiftLint, particularly in streamlining the code correction process during the build phase.

I am aware that running swiftlint autocorrect as part of the build script can lead to issues with Xcode's new build system and parallel tasks, such as the error "file was modified during build". However, it would be highly beneficial to have an option to enable autocorrect, either as a compiler flag, build setting, or through an environment variable, to ensure code corrections are made before the build process and not in parallel with other tasks.

This feature would greatly enhance SwiftLint's integration with Xcode, allowing for a more automated and efficient code review and correction process during development.
